### PR TITLE
Add "role" labels to cluster replicas

### PR DESCRIPTION
**Pull request by umanwizard**
This causes Kubernetes labels to be applied to the pods in replicas, allowing us to distinguish system-managed clusters from others. The labels are `system-critical` for `mz_system`, `system` for `mz_introspection`, and `user` for everything else.

### CODE REVIEW

File: src/adapter/src/catalog.rs [hunk 1]
```diff
@@ -1363,6 +1364,15 @@ pub struct ClusterReplica {
 }
 
 impl ClusterReplica {
+    pub fn role(&self) -> ReplicaRole {
+        // NOTE - These roles power monitoring systems. Do not change
+        // them without talking to the cloud or observability groups.
+        match self.name.as_str() {
+            "mz_system" => ReplicaRole::SystemCritical,
+            "mz_introspection" => ReplicaRole::System,
```
Comment (jkosh44) [inline src/adapter/src/catalog.rs:1372]: Is it possible to replace these with `SYSTEM_USER.name` and `INTROSPECTION_USER.name`?

**Review by jkosh44 — APPROVED** [review]

File: src/adapter/src/catalog.rs (hunk 1 above)
Comment (umanwizard) [reply src/adapter/src/catalog.rs:1372]: Yes, that's a good point

**Review by benesch — APPROVED** [review]
This is lovely, thank you!

**Comment by umanwizard** [comment]
This needs to change to look at the cluster name, rather than the replica name.

**Comment by benesch** [comment]
Also would be good to add a cloudtest test that locks in this behavior.

**Comment by umanwizard** [comment]
I'm making this into a function on `Cluster`, but I'm still passing it through the controller replica-scoped methods (e.g. `create_replica` and `provision_replica`) as that is the simplest way to write the code, and also means we'll have to change less places if we ever make this a per-replica property.

**Pull request merged**